appcelerator/titanium_mobile
# fix(android): hide/show progress dialog back-to-back puts it in bad state as if 8.1.1

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-27573

**Summary:**
Calling progress dialog's `hide()` and then immediately `show()` back-to-back prevents the dialog from ever being hidden again. Caused by the Java `onDismiss()` listener for previous dialog being invoked just after `show()` method call for the new dialog.

**Test 1:**
_(Verifies the above ticket's fix.)_
1. Build and run the code attached to [TIMOB-27573](https://jira.appcelerator.org/browse/TIMOB-27573) on Android.
2. Tap on the "Show Progress Dialog" button.
3. Verify that dialog is shown.
4. Verify that dialog is quickly hidden/shown 3 seconds later.
5. Verify that dialog is hidden 2 seconds. _(This used to fail.)_

---
**Test 2:**
_(Verifies progress dialog can be re-shown after auto-closed.)_
1. Build and run the code attached to [TIMOB-27309](https://jira.appcelerator.org/browse/TIMOB-27309) on Android.
2. Tap on the "Show Progress Dialog" button.
3. Wait 2 seconds for the progress dialog and its window to close.
4. Tap on the "Show Progress Dialog" button again.
5. Verify that the progress dialog is re-shown.
